### DISCUSSION TITLE
WebSocket Connection Issue: Working Locally but Failing After Deployment on Vercel with Next.js

### DISCUSSION THREAD

**Discussion by TusharKumawat1** [vercel/community#6695] [Help]
### Summary

My Next.js 14 application, which utilizes Socket.IO, functions correctly in my local environment. However, upon deployment using Vercel, I encounter an error stating "WebSocket connection to URL failed.

### Example

https://doodle-delight.vercel.app/

### Steps to Reproduce

Check console

**Comment by versecafe** [comment]
@TusharKumawat1 assuming you're following Socket IO's guide you can't deploy to Vercel due to using a custom server, there may be some work arounds but you'll also be stopped by the function limit of 300 seconds on pro

<img width="1320" alt="image" src="https://github.com/vercel/community/assets/147033096/ff1f4fdc-d5b2-4c33-bf8b-e640cb9ea2a9">
